v0.3.21 raised MSRV to 1.47.0

The readme lists MSRV 1.30, but patch release 0.3.21 raised that to 1.47.0 by using `Result::as_deref`. I have a project that uses 1.46, and don't mind if it needs to be raised, just thought I'd check first. Here's the error:

```
error[E0658]: use of unstable library feature 'inner_deref'
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/pkg-config-0.3.21/src/lib.rs:163:53
    |
163 |                         let crate_name = crate_name.as_deref().unwrap_or("sys");
    |                                                     ^^^^^^^^
    |
    = note: see issue #50264 <https://github.com/rust-lang/rust/issues/50264> for more information
```

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -159,8 +159,8 @@ impl fmt::Display for Error {
             } => {
                 match cause.kind() {
                     io::ErrorKind::NotFound => {
-                        let crate_name = std::env::var("CARGO_PKG_NAME");
-                        let crate_name = crate_name.as_deref().unwrap_or("sys");
+                        let crate_name =
+                            std::env::var("CARGO_PKG_NAME").unwrap_or_else(|_| "sys".to_owned());
                         let instructions = if cfg!(target_os = "macos") || cfg!(target_os = "ios") {
                             "Try `brew install pkg-config` if you have Homebrew.\n"
                         } else if cfg!(unix) {

